File: nativecrashcatch/src/main/java/com/flash/nativecatch/NativeHandler.java
package com.flash.nativecatch;

public class NativeHandler {

    public OnNativeCrashed onNativeCrashed;

    public void setOnNativeCrashed(OnNativeCrashed onNativeCrashed) {
        this.onNativeCrashed = onNativeCrashed;
    }

    public interface OnNativeCrashed {
        void onNativeCrashed();
    }

    static {
        System.loadLibrary("nativeCrashHandler-lib");
    }

    public native void createNativeException();

    public native void initNativeHandler(int pid);

    public void onNativeCrash() {
        if (onNativeCrashed != null) {
            onNativeCrashed.onNativeCrashed();
        }
    }
}

File: nativecrashcatch/src/main/java/com/flash/nativecatch/NativeHandler.kt
package com.flash.nativecatch

/**
 * Created by tianxiaolei on 2018/1/3.
 */
class NativeHandler {
    var onNativeCrashed: OnNativeCrashed? = null

    interface OnNativeCrashed {
        fun onNativeCrashed()
    }

    companion object {
        // Used to load the 'native-lib' library on application startup.
        init {
            System.loadLibrary("nativeCrashHandler-lib")
        }
    }

    external fun createNativeException(): Boolean

    external fun initNativeHandler(pid: Int)

    fun onNativeCrash() {
        onNativeCrashed?.onNativeCrashed()
    }
}
